src/extension.ts

```diff
@@ -61,11 +29,37 @@ export async function activate(context: vscode.ExtensionContext) {
 	const repository = new Repository(rootPath);
 	let repositoryInitialized = false;
 	let prManager: PullRequestManager;
+
 	repository.onDidRunGitStatus(async e => {
 		if (repositoryInitialized) {
 			return;
 		}
 
+		let remotes = repository.remotes.filter(remote => remote.host);
+		let remote = remotes.find(remote => remote.remoteName === 'origin');
```

Comment:
We may want to avoid relying on `origin` or probably making this a configuration (for example, `github.defaultRemoteName`). While working on VSCode code base, I use `upstream` for Microsoft/vscode and `rebornix` for my own fork, so it will fail if we check `origin`. 

It's a corner case I believe but all our code logic right now doesn't rely on `origin`.

Reply:
Yes, I agree, hardcoding `origin` is bad. Your logic right now relies on looking for a remote that is pointing to `github.com`; this logic finds either one that's called `origin` or the first remote if `origin` doesn't exist.

This means it will likely find `rebornix` for you before it finds `upstream`, if you cloned your fork first and then added the `upstream` remote. It's highly dependent on the clone/remote order. Definitely not ideal :P

On GHfVS, I think we look for remotes and then check whether the repository they point to is a fork, in which case we look up the fork parent and use that by default for PRs. I think we also let you switch between PR sources for a fork/upstream combo (i.e., you can see the PRs from upstream or you can see PRs from a fork, for long-lived project forks that have their own community)

The reason I don't want to scan for `github.com` is because github enterprise instances will work just as well with these features, and we can't really rely on looking at the name of the server. I probably need to add a check for a specific header that dotcom or enterprise always return, to make sure the server is usable (besides the whole "which remote do I use" thing)

A configuration option is probably good enough to start with to avoid these issues.